core/src/main/java/com/sap/ai/sdk/core/DestinationResolver.java

```diff
@@ -1,119 +1,57 @@
 package com.sap.ai.sdk.core;
 
-import static com.google.common.collect.Iterables.tryFind;
 import static com.sap.cloud.sdk.cloudplatform.connectivity.OnBehalfOf.TECHNICAL_USER_PROVIDER;
-import static com.sap.cloud.sdk.cloudplatform.connectivity.ServiceBindingDestinationOptions.forService;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.ObjectMapper;
 import com.sap.cloud.environment.servicebinding.api.DefaultServiceBindingAccessor;
-import com.sap.cloud.environment.servicebinding.api.DefaultServiceBindingBuilder;
+import com.sap.cloud.environment.servicebinding.api.ServiceBinding;
 import com.sap.cloud.environment.servicebinding.api.ServiceBindingAccessor;
 import com.sap.cloud.environment.servicebinding.api.ServiceBindingMerger;
 import com.sap.cloud.environment.servicebinding.api.ServiceIdentifier;
+import com.sap.cloud.environment.servicebinding.api.exception.ServiceBindingAccessException;
 import com.sap.cloud.sdk.cloudplatform.connectivity.HttpDestination;
 import com.sap.cloud.sdk.cloudplatform.connectivity.ServiceBindingDestinationLoader;
-import java.util.HashMap;
+import com.sap.cloud.sdk.cloudplatform.connectivity.ServiceBindingDestinationOptions;
 import java.util.List;
-import java.util.Optional;
-import java.util.function.Predicate;
 import javax.annotation.Nonnull;
-import javax.annotation.Nullable;
-import lombok.AccessLevel;
-import lombok.Getter;
-import lombok.NoArgsConstructor;
+import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import lombok.val;
 
 /** Utility class to resolve the destination pointing to the AI Core service. */
 @Slf4j
-@NoArgsConstructor(access = AccessLevel.PRIVATE) // utility class should not be instantiated
+@AllArgsConstructor
 class DestinationResolver {
-  static final String AI_CLIENT_TYPE_KEY = "URL.headers.AI-Client-Type";
-  static final String AI_CLIENT_TYPE_VALUE = "AI SDK Java";
+  @Nonnull private final ServiceBindingAccessor accessor;
 
-  @Getter(AccessLevel.PROTECTED)
-  @Nonnull
-  private static ServiceBindingAccessor accessor = DefaultServiceBindingAccessor.getInstance();
+  DestinationResolver() {
+    this(
+        new ServiceBindingMerger(
+            List.of(DefaultServiceBindingAccessor.getInstance(), new AiCoreServiceKeyAccessor()),
+            ServiceBindingMerger.KEEP_EVERYTHING));
```

Comment:
(Major/Question)

I see no cache in place, does that mean the service bindings are resolved again with every AI Core interaction?

Reply:
The default accessor will have a cache for the "normal" bindings. For the service key accessor, I now added a test that verifies files are only loaded once.